### PR TITLE
[Bug-fix] Prevent a SIT from transferring a user without a previous cohort

### PR DESCRIPTION
**Pull request by xswoodsx**
### Context

There's a [sentry error](https://sentry.io/organizations/dfe-teacher-services/issues/3353853344/?project=5748989&referrer=issue_alert-slack) where if a SIT can try to transfer a ppt from to the 2021 to their school, even if they don't have a school cohort for that year. This blocks them from doing it. We need them, at the minute, to create a full induction programme before they're able to do so. This is a quick fix, as agreed with Mili, Brian and Jo, that tells the user to go to support.

- Ticket: No ticket, but there is one in the backlog to handle this in a better way [828](https://dfedigital.atlassian.net/browse/CST-828)

### Changes proposed in this pull request
Redirecting to contact support if the school cohort does not have a FIP for the 2021 cohort year set up. 

Screenshot 2022-06-30 at 14 26 17

### CODE REVIEW

File: app/views/schools/transferring_participants/contact_support.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<% content_for :title, "This ECT’s training record needs to be transferred manually" %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+
+    <span class="govuk-caption-xl"><%= @school.name %></span>
+    <h1 class="govuk-heading-xl">This ECT’s training record needs to be transferred manually</h1>
+
+    <p class="govuk-body">Email our support tam at
+      <%= govuk_mail_to "continuing-professional-development@digital.education.gov.uk", "continuing-professional-development@digital.education.gov.uk" %>
```
Comment (jamesrose) [inline app/views/schools/transferring_participants/contact_support.html.erb:10]:
```suggestion
      <%= govuk_mail_to "continuing-professional-development@digital.education.gov.uk" %>
```

You don't need the second param here if they are the same value :)

File: app/helpers/transfer_participant_helper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module TransferParticipantHelper
+  def transfer_participant_in_cohort(cohort, school)
+    school_cohort = SchoolCohort.find_by(school:, cohort:)
+
+    if doing_fip?(school_cohort)
```
Comment (tonyheadford) [inline app/helpers/transfer_participant_helper.rb:7]:
```suggestion
    if school_cohort&.full_induction_programme?
```

File: app/helpers/transfer_participant_helper.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    else
+      contact_support_schools_transferring_participant_path
+    end
+  end
+
+private
+
+  def doing_fip?(school_cohort)
+    school_cohort.induction_programmes.any?(&:full_induction_programme?)
+  end
```
Comment (tonyheadford) [inline app/helpers/transfer_participant_helper.rb:18]:
```suggestion
```

**Review by tonyheadford — APPROVED** [review]
Suggestion to make it safer if school cohort not added and simplify a bit

File: app/helpers/transfer_participant_helper.rb (hunk 2 above)
Comment (xswoodsx) [reply app/helpers/transfer_participant_helper.rb:7]: Yeah, that's nicer :)

**Pull request merged**